packages/gatsby/src/services/initialize.ts

```diff
@@ -178,6 +179,26 @@ export async function initialize({
     })
   }
 
+  // Setup flags
+  if (config && config.flags) {
+    const flags = require(`../utils/flags`).default
+    // Get flags
+    const { validConfigFlags, message } = handleFlags(flags, config.flags)
+
+    //  set process.env for each flag
+    validConfigFlags.forEach(flag => {
+      process.env[flag.env] = `true`
+    })
+
+    //  track usage of feature
+    validConfigFlags.forEach(flag => {
+      telemetry.trackFeatureIsUsed(_.upperFirst(_.camelCase(flag.name)))
+    })
+
+    // Print out message.
+    reporter.info(message)
```

Comment:
if any flag is experimental, we should add a message that these flags are a bit more unstable.

see below

https://github.com/vercel/next.js/blob/450d4bd0f32a042fd452c81bc3850ec31306eab3/packages/next/next-server/server/config.ts#L73-L76